pkg/replication/replication.go

```diff
@@ -278,6 +288,20 @@ func (r replication) handleRenewalErrors(session kp.Session, renewalErrCh chan e
 
 // note: logging should be delegated somehow
 func (r replication) updateOne(node types.NodeName, done chan<- types.NodeName, quitCh <-chan struct{}, aggregateHealth *podHealth) {
+	defer func() {
+		select {
+		case done <- node:
+		case <-quitCh:
+		}
+	}()
+
+	select {
+	case <-r.quitCh:
+		return
+	case <-r.replicationCancelledCh:
```

Comment:
don't think you need to select on cancelledCh (because handleRenewalErrors will close `quitCh` after a cancellation